iml-wasm-components/iml-stratagem/src/inode_table.rs

```diff
@@ -0,0 +1,267 @@
+use bootstrap_components::bs_table::table;
+use futures::Future;
+use seed::{
+    error,
+    fetch::{FailReason, FetchObject, Request, RequestController},
+    prelude::*,
+    td, tr,
+};
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, PartialEq, Clone)]
+pub struct INode {
+    counter_name: String,
+    count: u32,
+}
+
+#[derive(Default, Debug)]
+pub struct Model {
+    inodes: Option<Vec<INode>>,
+    destroyed: bool,
+    cancel: Option<futures::sync::oneshot::Sender<()>>,
+    request_controller: Option<RequestController>,
+}
+
+#[derive(Clone, Debug)]
+pub enum Msg {
+    FetchInodes,
+    InodesFetched(FetchObject<InfluxResults>),
+    OnFetchError {
+        msg: String,
+        fail_reason: FailReason,
+    },
+    Destroy,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxSeries {
+    #[serde(skip)]
+    name: String,
+
+    #[serde(skip)]
+    columns: Vec<String>,
+
+    values: Vec<(String, String, u32)>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxResult {
+    #[serde(skip)]
+    statement_id: u16,
+    series: Option<Vec<InfluxSeries>>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxResults {
+    results: Vec<InfluxResult>,
+}
+
+pub fn update(msg: Msg, model: &mut Model, orders: &mut Orders<Msg>) {
+    if model.destroyed {
+        return;
+    }
+
+    match msg {
+        Msg::FetchInodes => {
+            model.cancel = None;
+
+            let (fut, request_controller) = fetch_inodes();
+            model.request_controller = request_controller;
+            orders.skip().perform_cmd(fut);
+        }
+        Msg::InodesFetched(fetch_object) => match fetch_object.response() {
+            Ok(response) => {
+                let data: InfluxResults = response.data;
+                model.inodes = data
```

Comment:
I was able to clean this up quite a bit:

```rust
data.results
                        .drain(..)
                        .take(1)
                        .filter_map(|result| result.series)
                        .flatten()
                        .take(1)
                        .map(|v| v.values)
                        .flatten()
                        .map(|(_, counter_name, count)| INode {
                            counter_name,
                            count,
                        })
                        .collect()
```

Since I used `drain` there are no more clones needed either.

Reply:
Thanks. I spent some time going through this and I understand it pretty well now. I agree this is cleaner.